t/testresults/00099/00099937-opensuse-13.1-DVD-i586-Build0091-kde/details-installer_timezone.json

```diff
@@ -1 +1 @@
-{"execution_time":34,"results":[{"area":[{"diff":"installer_timezone-1-inst-timezone-diff0.png","h":708,"result":"ok","similarity":96,"w":759,"x":"261","y":"0"}],"needle":"inst-timezone","result":"ok","screenshot":"installer_timezone-1.png","tags":["inst-timezone"]}]}
\ No newline at end of file
+{"dents":0,"details":[{"area":[{"diff":"installer_timezone-1-inst-timezone-diff0.png","h":708,"result":"ok","similarity":96,"w":759,"x":"261","y":"0"}],"needle":"inst-timezone","result":"ok","screenshot":"installer_timezone-1.png","tags":["inst-timezone"]}],"execution_time":34,"result":"ok"}
```

Comment:
So all of these `JSON` files now have `dents` and `result` in them, but I see no difference in the tests. Nothing seems to check what fields are in there now? Can we check that we're actually producing the expected format?

Reply:
For instance seeing that `dents, `result`, `details` are present, what they contain, and fail if they're missing.

Reply:
This might align nicely with adding documentation for the format, i.e. [poo#64117](https://progress.opensuse.org/issues/64117)

Reply:
`dents` and `result` are not used in openQA, and they do not affect the code used to parse the details file.
When job is running, the `dents` and `details` are not in details file. so the details will be an empty array when getting the job modules details information e.g. loading the test details page.

Reply:
> dents are not used in openQA

but `git $ git grep '\<dents\>'` tells me something different:

```
lib/OpenQA/Parser/Format/JUnit.pm:            dents   => 0,
lib/OpenQA/Parser/Format/TAP.pm:                dents   => 0,
lib/OpenQA/Parser/Format/XUnit.pm:        $result->{dents}  = 0;
lib/OpenQA/Parser/Result/OpenQA.pm:has dents   => 0;
lib/OpenQA/Parser/Result/OpenQA.pm:        dents   => $self->dents,
lib/OpenQA/Parser/Result/OpenQA.pm:        dents   => $self->dents,
lib/OpenQA/Parser/Result/OpenQA.pm:                                                      dents => 3,
lib/OpenQA/Parser/Result/OpenQA.pm:    my $dents   = $result->dents();
lib/OpenQA/Parser/Result/OpenQA.pm:    $result->dents(4);
lib/OpenQA/Parser/Result/OpenQA.pm:and implements the following new ones: C<details()>, C<dents()>, C<result()>, C<name()> and C<test()>.
lib/OpenQA/Parser/Result/OpenQA.pm:Note: only C<details()>, C<dents()> and C<result()> are 'required' by openQA.
lib/OpenQA/Parser/Result/OpenQA.pm:                                                      dents => 3,
lib/OpenQA/Parser/Result/OpenQA.pm:                                                      dents => 3,
lib/OpenQA/Parser/Result/OpenQA.pm:                                          dents => 3,
lib/OpenQA/Parser/Result/OpenQA.pm:    # { details => [], dents=> 3, result=>'ok' }
lib/OpenQA/Schema/Result/JobModules.pm:    $result = 'softfailed' if ($r->{dents} && $result eq 'passed');
t/10-jobs.t:    $job->update_module('d', {result => 'ok', details => [], dents => 1});
t/10-jobs.t:    $job->update_module('c', {result => 'ok', details => [], dents => 1});
t/10-jobs.t:    $job->update_module('c', {result => 'ok', details => [], dents => 1});
t/10-jobs.t:    $job->update_module('b', {result => 'ok', details => [], dents => 1});
t/10-jobs.t:    $job->update_module('c', {result => 'ok', details => [], dents => 1});
t/10-jobs.t:    $job->update_module('a', {result => 'ok', details => [], dents => 1});
t/10-jobs.t:    $job->update_module('a', {result => 'ok', details => [], dents => 1});
t/10-jobs.t:    $job->update_module('a', {result => 'ok', details => [], dents => 1});
t/30-test_parser.t:        dents   => 0,
t/30-test_parser.t:        'dents'   => 0,
t/dummy-isotovideo.sh:echo '{"dents":0,"details":[],"result":"ok"}' > testresults/result-fake.json
```

What do you mean with "not used"?

Reply:
Right, there's the parser framework (everything within `lib/OpenQA/Parser`). I don't know it well enough to say for sure whether it will not be affected but I would assume that it is not relevant here anyways because it comes in the "processing chain" before this change.

The other places should also be looked into as well but they are hopefully tested/covered in other tests already.

Reply:
Thanks @Martchus reply.
I mean the `dents` and `result` in the details files are not used in openQA. The `dents` in JobModules.pm comes from the REST API calls. And I know little about the `Parser`, I assume the `dents` used in here is not by reading from the details file (Maybe also comes from REST API).
@Martchus   Could I think  that the word `processing chain` you mentioned means the worker call a REST API, such as this `http://127.0.0.1:9526/api/v1/jobs/466/status`?

Reply:
I'm not familiar with the details of the parser framework. Hence I only used the vague term "processing chain". I'm afraid it is quite some effort to reverse-engineer the parser framework.

Reply:
What you can do is find tests that use the different parsers, I assume e.g. some "ltp tests" for openSUSE, and run them as tests against the complete stack, e.g. os-autoinst+openQA+ltp-tests